recipe/setup_py-win.patch

```diff
@@ -0,0 +1,80 @@
+--- eccodes-2.12.3-Source.orig/python3/setup.py.in	2019-02-27 09:30:50.000000000 -0300
++++ eccodes-2.12.3-Source/python3/setup.py.in	2019-04-08 11:03:10.313350958 -0300
+@@ -24,45 +24,45 @@
+                              '@CMAKE_CURRENT_SOURCE_DIR@/../src',
+                              numpy_include],
+                library_dirs=['@CMAKE_BINARY_DIR@/lib'],
+-               runtime_library_dirs=[],
++            #    runtime_library_dirs=[],
+                libraries=['eccodes'],
+                extra_compile_args=extra_compile_args,
+                extra_objects=[])
+ 
+ shared_libs='@BUILD_SHARED_LIBS@'
+-if shared_libs == 'OFF':
++# if shared_libs == 'OFF':
```

Comment:
`shared_libs` will be `ON`, so there's no need to patch the lines below as they are not run. Only the line 8 needs to be patched

Reply:
Thanks! Let me try that.

Reply:
Wait. I'm lost now :smile: line 8 of this file is a comment, right?

Reply:
I mean a patch like below,
```diff
diff --git a/python/CMakeLists.txt b/python/CMakeLists.txt
index 33aa1f0..41a1972 100644
--- a/python/CMakeLists.txt
+++ b/python/CMakeLists.txt
@@ -37,7 +37,7 @@ if( HAVE_PYTHON )
     set( _gribapi_swig "gribapi/_gribapi_swig${CMAKE_SHARED_LIBRARY_SUFFIX}" )
     # Build the extension module for use in build tree with RPATH pointing to the build tree
     add_custom_command( OUTPUT ${_gribapi_swig}
-                        COMMAND ${PYTHON_EXECUTABLE} setup.py build_ext --inplace --rpath ${CMAKE_BINARY_DIR}/lib
+                        COMMAND ${PYTHON_EXECUTABLE} setup.py build_ext --inplace
                         DEPENDS grib_interface.h grib_interface.c swig_wrap_numpy.c setup.py.in eccodes )
     add_custom_target(eccodes_build_swig_wrapper ALL DEPENDS ${_gribapi_swig})
 
@@ -56,8 +56,8 @@ if( HAVE_PYTHON )
 
     # Build the extension module for use in install tree with RPATH pointing to install tree
     install(CODE "message(\"Building Python extension modules:
-${PYTHON_EXECUTABLE} setup.py build_ext --rpath ${CMAKE_INSTALL_PREFIX}/${INSTALL_LIB_DIR}\")
-                  execute_process(COMMAND ${PYTHON_EXECUTABLE} setup.py build_ext --rpath ${CMAKE_INSTALL_PREFIX}/${INSTALL_LIB_DIR}
+${PYTHON_EXECUTABLE} setup.py build_ext\")
+                  execute_process(COMMAND ${PYTHON_EXECUTABLE} setup.py build_ext
                                   WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR})")
 
     # Library installation directory override
diff --git a/python/setup.py.in b/python/setup.py.in
index f54d428..c773e37 100644
--- a/python/setup.py.in
+++ b/python/setup.py.in
@@ -24,7 +24,7 @@ attdict = dict(sources=['@CMAKE_CURRENT_SOURCE_DIR@/swig_wrap_numpy.c',
                              '@CMAKE_CURRENT_SOURCE_DIR@/../src',
                              numpy_include],
                library_dirs=['@CMAKE_BINARY_DIR@/lib'],
-               runtime_library_dirs=[],
+               #runtime_library_dirs=[],
                libraries=['eccodes'],
                extra_compile_args=extra_compile_args,
                extra_objects=[])
diff --git a/python3/CMakeLists.txt b/python3/CMakeLists.txt
index 33aa1f0..41a1972 100644
--- a/python3/CMakeLists.txt
+++ b/python3/CMakeLists.txt
@@ -37,7 +37,7 @@ if( HAVE_PYTHON )
     set( _gribapi_swig "gribapi/_gribapi_swig${CMAKE_SHARED_LIBRARY_SUFFIX}" )
     # Build the extension module for use in build tree with RPATH pointing to the build tree
     add_custom_command( OUTPUT ${_gribapi_swig}
-                        COMMAND ${PYTHON_EXECUTABLE} setup.py build_ext --inplace --rpath ${CMAKE_BINARY_DIR}/lib
+                        COMMAND ${PYTHON_EXECUTABLE} setup.py build_ext --inplace
                         DEPENDS grib_interface.h grib_interface.c swig_wrap_numpy.c setup.py.in eccodes )
     add_custom_target(eccodes_build_swig_wrapper ALL DEPENDS ${_gribapi_swig})
 
@@ -56,8 +56,8 @@ if( HAVE_PYTHON )
 
     # Build the extension module for use in install tree with RPATH pointing to install tree
     install(CODE "message(\"Building Python extension modules:
-${PYTHON_EXECUTABLE} setup.py build_ext --rpath ${CMAKE_INSTALL_PREFIX}/${INSTALL_LIB_DIR}\")
-                  execute_process(COMMAND ${PYTHON_EXECUTABLE} setup.py build_ext --rpath ${CMAKE_INSTALL_PREFIX}/${INSTALL_LIB_DIR}
+${PYTHON_EXECUTABLE} setup.py build_ext\")
+                  execute_process(COMMAND ${PYTHON_EXECUTABLE} setup.py build_ext
                                   WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR})")
 
     # Library installation directory override
diff --git a/python3/setup.py.in b/python3/setup.py.in
index 57dc654..34ac5c4 100644
--- a/python3/setup.py.in
+++ b/python3/setup.py.in
@@ -24,7 +24,7 @@ attdict = dict(sources=['@CMAKE_CURRENT_SOURCE_DIR@/swig_wrap_numpy.c',
                              '@CMAKE_CURRENT_SOURCE_DIR@/../src',
                              numpy_include],
                library_dirs=['@CMAKE_BINARY_DIR@/lib'],
-               runtime_library_dirs=[],
+               #runtime_library_dirs=[],
                libraries=['eccodes'],
                extra_compile_args=extra_compile_args,
                extra_objects=[])
```

Reply:
Thanks! I'll give it a try. (Got side tracked and forgot about this one.)